contracts/Turnstone.vy

```diff
@@ -0,0 +1,105 @@
+# @version 0.3.3
+"""
+@title Turnstone
+@author Volume.Finance
+"""
+
+MAX_VALIDATORS: constant(uint256) = 128
+MAX_BATCH: constant(uint256) = 128
+POWER_THRESHOLD: constant(uint256) = 2_863_311_530 # 2/3 of 2^32
```

Comment:
I'm not sure I understand the logic involving this variable. Are we assuming that externally each validator is being assigned a power that is a fraction of 2^32? If so could you add some documentation speaking to this assumption?

Reply:
I am not really a fan of this as it's really confusing. Could we use the total_power from the ValsetArgs?

Reply:
This is a threshold. All validators don't need to sign and if the validators whose power sum is more than enough power threshold, the signature will work. Some validators may be absent.

Reply:
Could you add a comment in the code indicating why 2_863_311_530 _specifically_ is the threshold, or change the threshold to be calculated from the Valset?

As currently written I don't follow what keeps this number in sync with the outside world. Does the caller of the update API ensure that `total_power` is 2^32? Is that something we check in this contract?

Reply:
I brought this from Gravity.sol of sommelier. We can update this value with Paloma logic.

Reply:
We should have Jehan review this as well. I'll ask him to come take a look.

Reply:
This has confused people before. The way this works is that validator powers are normalized to sum to 2^32 in every valset update. This is why the threshold is expressed as a fraction of 2^32. 

An easy way to think about this is to imagine that validator powers are normalized to 100, i.e. percentages. Then the percentages are summed up to get to 66%, i.e. 2/3s of 100. Of course, using 100 would be far too little precision, so we use a smaller number.

We could use max 2^256 instead of 2^32, but this is way more precision than necessary, so I think that 2^32 was chosen since 32 bit integers are more than big enough.